### PR TITLE
feat(components): add Text component to display colored text

### PR DESCRIPTION
**Pull request by peter-sanderson**
Add the `Text` component to be used in the future when we need text with variants and other typographic/UI features.

image

### CODE REVIEW

File: packages/suite/src/views/dashboard/components/T2B1PromoBanner.tsx [hunk 1]
```diff
@@ -272,7 +255,7 @@ export const T2B1PromoBanner = () => {
                     <TrezorSafe3Logo image="TREZOR_SAFE_PROMO_LOGO" />
                     <NextGenerationText>
```
Comment (jvaclavik) [inline packages/suite/src/views/dashboard/components/T2B1PromoBanner.tsx]: What about not use `font-size` in `NextGenerationText` and put new prop to `Text`? 
It should have these styles:
image

WDYT?

Comment (peter-sanderson) [reply packages/suite/src/views/dashboard/components/T2B1PromoBanner.tsx]: I have added the typography to the `<Text>` however I would need a help with applying it.

File: packages/components/src/components/typography/Text/Text.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import styled from 'styled-components';
+import { UIVariant } from '../../../config/types';
+import { CSSColor, Color, Colors, TypographyStyle, typography } from '@trezor/theme';
+import { ReactNode } from 'react';
+
+export type TextVariant = Extract<UIVariant, 'primary' | 'info' | 'warning' | 'destructive'>;
+
+const variantColorMap: Record<TextVariant, Color> = {
+    primary: 'borderSecondary',
```
Comment (jvaclavik) [inline packages/components/src/components/typography/Text/Text.tsx]: `textPrimaryDefault`

File: packages/components/src/components/typography/Text/Text.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    info: 'textAlertBlue',
+    warning: 'textAlertYellow',
+    destructive: 'borderAlertRed',
```
Comment (jvaclavik) [inline packages/components/src/components/typography/Text/Text.tsx]: `textAlertRed`

**Review by jvaclavik — APPROVED** [review]

**Pull request merged**